### PR TITLE
support ls-remote command and git+ssh protocol

### PR DESCRIPTION
**Pull request by jakecoffman**
The integration test uses a Docker container that doesn't have `ssh` installed so the commands would fail normally. This is a simpler setup than building a Docker network and achieves the same result. If the `docker build` fails then the test fails!

### CODE REVIEW

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -23,3 +23,10 @@ jobs:
 
     - name: Test
       run: go test -v ./...
+
+  integration-test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Integration test
+        run: docker build -f test/Dockerfile .
```
Comment (jeffwidman) [inline .github/workflows/go.yml:32]:
```suggestion
        # `docker build` _is_ the test, if it fails, the test has failed.
        run: docker build -f test/Dockerfile .
```

**Review by jeffwidman — APPROVED** [review]

**Comment by deivid-rodriguez** [comment]
Did we try this against https://github.com/dependabot/dependabot-core/issues/7258? I'm afraid it's now going to start "succeeding" now but generating incorrect PRs since it will "private repo syntax" for the dependency using ssh.

**Comment by jakecoffman** [comment]
@deivid-rodriguez I haven't cut a release of git-shim and bumped core yet, so all is good still. I'll repro it and see if this changes anything. Thanks for the heads up!

**Comment by deivid-rodriguez** [comment]
Ok, great! I mean it would not be the end of the world, but generally I consider a completely broken update better than a successful update with the wrong PR contents.

I did try changing git-shim last week to do this, but I think I didn't build the binary properly or I didn't put it at the right path, because it seemed to have no effect

I did change our `insteadOf` git config commands in core to re-write git+ssh urls too and that did cause the issue I'm taking about here, so I didn't proceed with making that change.

**Comment by jakecoffman** [comment]
Just tried locally on main with the updated shim, it still fails, but on a `git fetch`

```
updater | Progress: resolved 1009, reused 1008, downloaded 0, added 0
updater |  ERROR  Command failed with exit code 128: git fetch --depth 1 origin 32e8e38c7fcd935241b9baab71bb432fd9b166ed
updater | error: cannot run ssh: No such file or directory
updater | fatal: unable to fork
updater | 
updater | pnpm: Command failed with exit code 128: git fetch --depth 1 origin 32e8e38c7fcd935241b9baab71bb432fd9b166ed
updater | error: cannot run ssh: No such file or directory
updater | fatal: unable to fork
updater |     at makeError (/home/dependabot/.cache/node/corepack/pnpm/8.3.1/dist/pnpm.cjs:24230:17)
updater |     at handlePromise (/home/dependabot/.cache/node/corepack/pnpm/8.3.1/dist/pnpm.cjs:24801:33)
updater |     at runMicrotasks (<anonymous>)
updater |     at processTicksAndRejections (node:internal/process/task_queues:96:5)
updater |     at async gitFetcher (/home/dependabot/.cache/node/corepack/pnpm/8.3.1/dist/pnpm.cjs:113224:11)
updater |     at async fetcher (/home/dependabot/.cache/node/corepack/pnpm/8.3.1/dist/pnpm.cjs:125583:16)
updater |     at async run (/home/dependabot/.cache/node/corepack/pnpm/8.3.1/dist/pnpm.cjs:124970:23)
```

No amount of argument rewriting can fix that since the URL isn't an argument. Looks like the upstream fix is definitely the way to go.

**Pull request merged**